code/modules/mob/say.dm

```diff
@@ -77,6 +87,9 @@
 	var/spanned = say_quote(message)
 	var/source = "<span class='game'><span class='prefix'>DEAD:</span> <span class='name'>[name]</span>[alt_name]"
 	var/rendered = " <span class='message'>[emoji_parse(spanned)]</span></span>"
+	for(var/datum/deadchat_listener/listener in GLOB.deadchat_listeners)
+		listener.message_event(src.client.ckey, message)

```

Comment:
I feel like signals could make this way nicer

Reply:
right now I am looking into making this into a component, will see if I manage to make it work.